completion/available/gradle.completion.bash

```diff
@@ -262,22 +278,30 @@ _gradle() {
 			__gradle-set-files-checksum
 
 			# The cache key is md5 sum of all gradle scripts, so it's valid if it exists.
-			if [[ -f $cache_dir/$cache_name.md5 ]]; then
-				local cached_checksum="$(cat $cache_dir/$cache_name.md5)"
+			if [[ -f "$cache_dir/$cache_name.md5" ]]; then
+				local cached_checksum
+				cached_checksum="$(cat "$cache_dir/$cache_name.md5")"
 				local -a cached_tasks
 				if [[ -z $cur ]]; then
-					cached_tasks=($(cat $cache_dir/$cached_checksum))
+					while IFS='' read -r line; do
+						cached_tasks+=("$line")
+					done < "$cache_dir/$cached_checksum"
 				else
-					cached_tasks=($(grep "^$cur" $cache_dir/$cached_checksum))
+					while IFS='' read -r line; do
+						cached_tasks+=("$line")
+					done < <(grep "^$cur" "$cache_dir/$cached_checksum")
 				fi
-				COMPREPLY=($(compgen -W "${cached_tasks[*]}" -- "$cur"))
+				while IFS='' read -r line; do
+					COMPREPLY+=("$line")
+				done < <(compgen -W "${cached_tasks[*]}" -- "$cur")
 			else
 				__gradle-notify-tasks-cache-build
 			fi
 
 			# Regenerate tasks cache in the background
-			if [[ $gradle_files_checksum != "$(cat $cache_dir/$cache_name.md5)" || ! -f $cache_dir/$gradle_files_checksum ]]; then
-				$(__gradle-generate-tasks-cache 1>&2 2> /dev/null &)
+			if [[ $gradle_files_checksum != "$(cat "$cache_dir/$cache_name.md5")" ||
+			! -f "$cache_dir/$gradle_files_checksum" ]]; then
+				__gradle-generate-tasks-cache &> /dev/null &
```

Comment:
Background process should be launched from a subshell.

```suggestion
				(__gradle-generate-tasks-cache &> /dev/null &)
```

If the job is backgrounded in the main shell process, the job will be registered to the job list of the main shell and managed by the job control. It would produce messages like

```bash
[1] <process id>
[1]+  Done                       __gradle-generate-tasks-cache &> /dev/null
```